### PR TITLE
Add `Observable` documentation and `Observable.from_pauli_string_collections` method

### LINKED ISSUE #972: Create an `Observable` by specifying `PauliStringCollection`s
There is currently no way to dictate how an `Observable` is partitioned into qubit-wise commuting sets. Right now, they are partitioned via a greedy algorithm.

https://github.com/unitaryfund/mitiq/blob/11c6c680f6bf8a6ebcbb236554003fa934998370/mitiq/observable/observable.py#L29-L34

One may wish to specify exactly how `PauliString`s are grouped and measured for more control when using Mitiq. We should support this.

### LINKED ISSUE #994: Add Observables to new docs
Issue Description
-----------------

As a part of #908, and as specified by the [RFC](https://docs.google.com/document/d/13un5TZPknSOhmOBkrL2rsofjGfdp2jDnd-DywLpGFPc/edit#heading=h.mckr847qq60k) we want to add a general concepts section to the docs about Observables.

### PR DESCRIPTION
**Pull request by rmlarose**
Description
-----------

Closes #972 and closes #994.

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

Tips
----

- If the validation check fails:

    1. Run `make check-types` (from the root directory of the repository) and fix any [mypy](https://mypy.readthedocs.io/en/stable/) errors.

    2. Run `make check-style` and fix any [flake8](http://flake8.pycqa.org) errors.

    3. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html) autoformatter.

  For more information, check the [Mitiq style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines).
  
- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

### CODE REVIEW

File: docs/source/guide/guide-observables.myst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+```{code-cell} ipython3
+from mitiq.observable.pauli import PauliStringCollection
+
+group1 = PauliStringCollection(pauli1)
+group2 = PauliStringCollection(pauli2, pauli3)
+
+obs = Observable.from_pauli_string_collections(group1, group2)
+```
+
+To see the matrix of the observable:
```
Comment (andreamari) [inline docs/source/guide/guide-observables.myst:75]:
```suggestion
To see the (potentially very large) matrix representation of the observable:
```

File: docs/source/guide/guide-observables.myst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+```{code-cell} ipython3
+from mitiq import zne
+```
+
+```{code-cell} ipython3
+executor = Executor(mitiq_cirq.compute_density_matrix)
+
+zne_value = zne.execute_with_zne(circuit, executor, obs)
+print("ZNE value:", zne_value)
```
Comment (andreamari) [inline docs/source/guide/guide-observables.myst:138]:
```suggestion
print(f"ZNE value: {zne_value :g}")
```
This should print the complex number without parenthesis and without too many digits.

File: docs/source/guide/guide-observables.myst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  language: python
+  name: python3
+---
+
+# Observables
+
++++
+
+The `mitiq.Observable` class is a way to represent an observable as a linear combination of Pauli strings. This class is used to compute expectation values which are mitigated by techniques in Mitiq.
+
```
Comment (andreamari) [inline docs/source/guide/guide-observables.myst:19]: We could add a comment saying that using explicit observables is not the only way of using Mitiq. We still support the usual simplified way. E.g. something like this, but feel free to reword:
```
Note however that Mitiq can also be used without explicitly defining a `mitiq.Observable` but by implicitly introducing the observable in the body of an `executor` function that directly returns an expectation value as a `float`.
```

**Review by andreamari — APPROVED** [review]
LGTM! 
Left a few suggestions

File: docs/source/guide/guide-observables.myst (hunk 3 above)
Comment (rmlarose) [reply docs/source/guide/guide-observables.myst]: After trying for a bit, I couldn't come to a clean way of adding this statement. Note that it is mentioned at the end of the guide (where to me seems more appropriate):

"If you do not provide an observable, the `executor` must compute and return the expectation value to mitigate."

I did change "This class is used to compute expectation values" to "This class *can be* used to compute expectation values" at the beginning, though.

**Pull request merged**